tests/e2e/idps_test.go

```diff
@@ -763,6 +761,193 @@ var _ = Describe("TF Test", func() {
+							},
+							Timeout: 7,
+						}
+						_, err = openshift.OcLogin(*ocAtter)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Delete ldap idp by OCM API")
+						cms.DeleteIDP(ci.RHCSConnection, clusterID, idpID.ID)
+						_, err = cms.RetrieveClusterIDPDetail(ci.RHCSConnection, clusterID, idpID.ID)
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).Should(ContainSubstring(
+							"Identity provider ID '%s' for cluster '%s' not found", idpID.ID, clusterID),
+						)
+
+						By("Re-apply the idp resource")
+						Expect(idpService.ldap.Apply(idpParam, false)).To(Succeed())
+
+						By("re-login with the ldpa idp username/password")
+						_, err = openshift.OcLogin(*ocAtter)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Delete ldap idp from existing cluster")
+						defer func() {
+							err := idpService.ldap.Destroy()
+							Expect(err).ToNot(HaveOccurred())
+						}()
+
+					})
+			})
+
+			Context("Author:smiron-Medium-OCP-65816 @OCP-65816 @smiron", func() {
+				It("OCP-65816 - try to restore/duplicate an existing IDP",
+					ci.Day2, ci.Medium, ci.FeatureIDP, func() {
+
+						gitLabIdpName = "OCP-65816-gitlab-idp-reconcil"
+
+						By("Create gitlab idp for existing cluster")
+						idpParam := &exe.IDPArgs{
+							ClusterID:    clusterID,
+							Name:         gitLabIdpName,
+							ClientID:     gitlabIDPClientId,
+							ClientSecret: gitlabIDPClientSecret,
+							URL:          con.GitLabURL,
+						}
+						Expect(idpService.gitlab.Apply(idpParam, false)).To(Succeed())
+						idpID, _ := idpService.gitlab.Output()
+
+						By("Delete gitlab idp using OCM API")
+						cms.DeleteIDP(ci.RHCSConnection, clusterID, idpID.ID)
+						_, err = cms.RetrieveClusterIDPDetail(ci.RHCSConnection, clusterID, idpID.ID)
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).Should(ContainSubstring(
+							"Identity provider ID '%s' for cluster '%s' not found", idpID.ID, clusterID),
+						)
+
+						By("Create gitlab idp using OCM api with the same parameters")
+						requestBody, _ := gitlabIdpOcmAPI.Type("GitlabIdentityProvider").
+							Name(gitLabIdpName).
+							Gitlab(cmv1.NewGitlabIdentityProvider().
+								ClientID(gitlabIDPClientId).
+								ClientSecret(gitlabIDPClientSecret).
+								URL(con.GitLabURL)).
+							MappingMethod("claim").
+							Build()
+						res, err := cms.CreateClusterIDP(ci.RHCSConnection, clusterID, requestBody)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(res.Status()).To(Equal(http.StatusCreated))
+
+						By("Re-apply gitlab idp using tf manifests with same ocm api args")
+						err = idpService.gitlab.Apply(idpParam, false)
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).Should(ContainSubstring(
+							"Identity Provider Name\n%s already exists", gitLabIdpName),
+						)
+
+						By("Delete gitlab idp from existing cluster")
+						defer cms.DeleteIDP(ci.RHCSConnection, clusterID, res.Body().ID())
```

Comment:
Put defer to last step is meaningless. Put it to where it is created.

Reply:
May be put in to AfterEach @xueli181114 ?

Reply:
I think AfterEach will cause issues because the test cases are independent

Reply:
fixed